pkg/scheduler/testing/utils.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeSelectorWrapper wraps a NodeSelector inside.
+type NodeSelectorWrapper struct{ v1.NodeSelector }
```

Comment:
maybe Builder is a more appropriate name for all these structs. And `builders.go` as filename.

Reply:
Sounds good.

Reply:
IMO `builder` doesn't sound more descriptive than `wrapper` (please correct me if anybody thing it's much better).

But I do like the idea to rename `utils.go` to `wrappers.go`/`builders.go`.

Reply:
A wrapper would be something that adds more fields that don't exist in the original struct. This is basically holding an object until you get it from `Obj`. Hence, it knows how to build the object.